app/helpers/markdown_form_builder.rb

```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+class MarkdownFormBuilder < ActionView::Helpers::FormBuilder
+  def markdown(method, args = {})
+    # Adopt simple form builder to work with form_for
+    @attribute_name = method
+    @input_html_options = args[:input_html]
+
+    @template.capture do
+      @template.tag.div(class: 'markdown-editor', data: {behavior: 'markdown-editor-wrapper'}) do
+        @template.concat form_textarea
+        @template.concat @template.tag.div(data: {behavior: 'markdown-editor-widget'})
+      end
+    end
+  end
+
+  private
+
+  def form_textarea
+    @template.text_area @object_name, @attribute_name,
+    **(@input_html_options || {}),
+      id: label_target,
+      data: {behavior: 'markdown-form-input'}
+  end
```

Comment:
This text are might be visible when the editor contains some text:

<img width="593" alt="Bildschirmfoto 2024-04-12 um 01 06 19" src="https://github.com/openHPI/codeocean/assets/7300329/f59894e8-c0c5-4d41-919d-1542f7077b93">

In the video, I am hiding the text area by adding and removing bootstrap's `d-none` class. Those areas highlighted in the previous screenshot are hidden and revealed.

https://github.com/openHPI/codeocean/assets/7300329/4c753b8b-e137-4274-8354-063b85c7e2aa

Reply:
Yes, I can see this too. I've hidden the `textarea` elements. With this the wrapper element is not needed anymore so the markdown could be simplified a bit. (See last commit. It can be squashed later.)

I think the intention behind not hiding it was to make the field still editable without javascript. But this situation is  probably a rare edge case...

Reply:
Perfect, thanks! This works as expected and I think that we can accept the JavaScript-only usage scenario (there are a bunch of other issues when JavaScript is not available).